components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/utils/RegistryPersistenceUtil.java

```diff
@@ -1219,6 +1219,11 @@ public static String getOpenAPIDefinitionFilePath(String apiName, String apiVers
                 + RegistryConstants.PATH_SEPARATOR + apiName + RegistryConstants.PATH_SEPARATOR + apiVersion
                 + RegistryConstants.PATH_SEPARATOR;
     }
+
+    public static String getAsyncAPIDefinitionFilePath(String apiName, String apipVersion, String apiProvider) {
```

Comment:
Let's not use this . use `String resourcePath = GovernanceUtils.getArtifactPath(registryType, apiId);` to get the path

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/10626